e-commerce-project/src/app/components/login-page/login-page.component.html

```diff
@@ -1,8 +1,18 @@
 <h3>Welcom to login page!</h3>
 <p>Enter your username and password. Then push Submit button.</p>
 
-<form (ngSubmit)="toSubmit(username.value, password.value)">
-	<input #username type="text" name= 'username' placeholder="Username"><br><br>
-	<input #password type="password" name="pass" placeholder="Password"><br><br>
-	<button type="submit">Submit</button>
-</form>
\ No newline at end of file
+<form [formGroup]="loginForm" (ngSubmit)="toSubmit()">
+
```

Comment:
1) we just need to add `[formGroup]=`  inside  `<form` 
https://angular.io/guide/reactive-forms#step-2-associating-the-formgroup-model-and-view
2) The `loginForm` is defined inside component below